lib/httpserver/httpserver.go

```diff
@@ -32,8 +32,9 @@ var (
 	metricsAuthKey   = flag.String("metricsAuthKey", "", "Auth key for /metrics. It overrides httpAuth settings")
 	pprofAuthKey     = flag.String("pprofAuthKey", "", "Auth key for /debug/pprof. It overrides httpAuth settings")
 
-	disableResponseCompression  = flag.Bool("http.disableResponseCompression", false, "Disable compression of HTTP responses for saving CPU resources. By default compression is enabled to save network bandwidth")
-	maxGracefulShutdownDuration = flag.Duration("http.maxGracefulShutdownDuration", 7*time.Second, "The maximum duration for graceful shutdown of HTTP server. "+
+	disableResponseCompression = flag.Bool("http.disableResponseCompression", false, "Disable compression of HTTP responses for saving CPU resources. By default compression is enabled to save network bandwidth")
+	// 7 is too small for online, and no argument can change this, make it bigger
+	maxGracefulShutdownDuration = flag.Duration("http.maxGracefulShutdownDuration", 120*time.Second, "The maximum duration for graceful shutdown of HTTP server. "+
```

Comment:
120 seconds timeout is too big for graceful shutdown of http server, since this may increase VM service outage on restart from a few seconds to up to 2 minutes. I improved error message for graceful shutdown timeout in the commit 01c17092e1a9f260ca77c90936efab671535c10a , so users could adjust `-http.maxGracefulShutdownDuration` flag value if they see this error.

Reply:
ok

Reply:
FYI, there was a bug in `lib/httpserver`, which could increase graceful shutdown duration for http server to more than a minute. This should be fixed in https://github.com/VictoriaMetrics/VictoriaMetrics/commit/787fcfba0ce361b728fb639496dd24994f4e7cd9 . Now typical graceful shutdown duration should take less than 5 seconds. The duration can be longer if slow requests are processed during the shutdown.